conman/routes/models.py

```diff
@@ -136,6 +136,16 @@ def handle(self, request, path):
         # Deal with the request
         return handler.handle(request, path)
 
+    @property
+    def level(self):
+        """Fetch the one-indexed 'level' of this item in the URL tree."""
+        return self.url.count('/')
+
+    @level.setter
+    def level(self, new_value):
+        """Silently fails to allow queryset annotation to work."""
+        pass
```

Comment:
Are you sure the annotation is used instead of the property if it exists? Or does that not matter?

Reply:
It's not that the annotation is used, exactly, it's that when an annotated object is instantiated, the attribute is set. Without this, and with the `@property`, a "can't set attribute" error is raised.

Reply:
I see - the annotation is used for sql-level operations like filtering, but the descriptor whenever we're in python.

Reply:
Exactly :)